api/src/test/java/io/strimzi/test/Statement.java

```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2018, Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.test;
+
+public abstract class Statement {
```

Comment:
Why do we need this class? or can you add java doc for it?

Reply:
Good point, it's used by Bracket class in the StrimziExtension. I'd like to remove this class, but currently I am not sure how. I want to discuss ideas about that.

Reply:
The original `Statement` was a concept from Junit 4. Basically the testing infrastructure would build up chains of `Statements` which could then be executed. By adding (or not) different statements you could do things before and/or after the tests. IDK how Junit 5 does things, I assume it has some other way of achieving the same thing. We should try to rewrite `Bracket` to use that API, whatever it is. `Bracket` is really about adapting `Statement` (which had `execute()`) into a `before()` and `after()`-style API using the template method pattern. Importantly `after()` gets called even if the JVM is killed.